org.eclipse.lsp4e/src/org/eclipse/lsp4e/ContentTypeToLanguageServerDefinition.java

```diff
@@ -48,8 +53,11 @@ public boolean isUserEnabled() {
 	}
 
 	public boolean isExtensionEnabled() {
-		return enablement != null ? enablement.evaluate() : true;
+		return isExtensionEnabled(null);
+	}
```

Comment:
same as above

Reply:
> I would prefer to delete those methods without argument, because then the developer is urged to provide a IDocument when using this method.

OK.

Reply:
@mickaelistria: Updated change. The evaluate method gets an URI and creates a document if its not pre-existing. It will be disposed after the check. Removed `isExtensionEnabled()` and `isEnabled()` call without arguments. Had to modify the tests according to the  removal.